### PR TITLE
Support Find References in umbrella apps

### PR DESCRIPTION
**Pull request by mattbaker**
Here's an attempt at addressing #55. Little heavy on the logging _but_ I could see it being nice if any problems occur. 

I ran this on a decently sized umbrella app with a compile time of 22 seconds. I confirmed that while "Find References" was waiting to acquire the lock, VSCode conveniently displays...

screen shot 2018-03-01 at 11 12 52 pm

Thoughts @JakeBecker?

### CODE REVIEW

**Comment by JakeBecker** [comment]
Looks good, though I'll take a closer look tomorrow. Since VS Code shows a nice loading indicator and can cancel requests if the user clicks away, this approach should work well 👍

**Comment by JakeBecker** [comment]
@mattbaker I made some small changes and I added tests. Please take a look, then I'm ready to merge. Seems to work great!

File: apps/language_server/lib/language_server/providers/references.ex [hunk 1]
```diff
@@ -50,8 +50,6 @@ defmodule ElixirLS.LanguageServer.Providers.References do
   defp add_arity({mod, fun}, %{scope: {fun, arity}, module: mod}), do: {mod, fun, arity}
   defp add_arity({mod, fun}, _env), do: {mod, fun, nil}
 
-  defp callers(nil), do: []
```
Comment (mattbaker) [inline apps/language_server/lib/language_server/providers/references.ex]: Nice, I think this was a holdover from some refactoring

**Comment by mattbaker** [comment]
Excellent stuff @JakeBecker, particularly the lock simplification, and thanks for the test examples. Looks great to me.

**Pull request merged**